modules/core/src/command_line_parser.cpp

```diff
@@ -134,22 +135,25 @@ void CommandLineParser::getByIndex(int index, bool space_delete, int type, void*
                 String v = impl->data[i].def_value;
                 if (space_delete == true) v = impl->cat_string(v);
 
-                // it is an error if we just got the default value here
-                if(v.empty())
-                    break;
-
+                // the key was neither specified nor has it a default value
+                if(v.empty()) {
```

Comment:
yes, but then how do you check for *mandatory* string parameters? I think that the point of the `CommandLineParser` is that it does the argument checking and error handling for you.

Then there is an inconsistency for positional arguments between calling `get<String>(0)` and `get<String>("@foo")` and between positional and keyword arguments in general.

However breaking existing applications is bad. Would you agree with adding an operation mode where the parser disallows empty default strings?